dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java

```diff
@@ -1,85 +1,168 @@
 
     /**
-     * <p>Returns a JSON representation of a page
-     * <p/>
-     * Usage: /page/{hostOrFolderIdentifier}
+     * Returns the metadata in JSON format of the objects that make up an HTML Page in the system.
+     * <pre>
+     * Format:
+     * http://localhost:8080/api/v1/page/json/{page-url}
+     * <br/>
+     * Example:
+     * http://localhost:8080/api/v1/page/json/about-us/locations/index
+     * </pre>
+     *
+     * @param request  The {@link HttpServletRequest} object.
+     * @param response The {@link HttpServletResponse} object.
+     * @param uri      The path to the HTML Page whose information will be retrieved.
+     * @return All the objects on an associated HTML Page.
      */
-    @GET
-    @JSONP
-    @Path("/page/{pageId}")
     @NoCache
+    @GET
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
-    public Map<String, String> list(@Context HttpServletRequest request, @PathParam("pageId") String pageId) {
-
-        Map<String, String> hash = Maps.newHashMap();
-
-
-        return hash;
+    @Path("/json/{uri: .*}")
+    public Response loadJson(@Context final HttpServletRequest request, @Context final
+    HttpServletResponse response, @PathParam("uri") final String uri) {
+        // Force authentication
+        final InitDataObject auth = webResource.init(false, request, true);
+        final User user = auth.getUser();
+        Response res = null;
+        try {
+            final PageView pageView = this.pageResourceHelper.getPageMetadata(request, response,
+                    user, uri);
+            final String json = this.pageResourceHelper.asJson(pageView);
+            final Response.ResponseBuilder responseBuilder = Response.ok(json);
+            responseBuilder.header("Access-Control-Expose-Headers", "Authorization");
+            responseBuilder.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, " +
+                    "Content-Type, " + "Accept, Authorization");
+            res = responseBuilder.build();
+        } catch (JsonProcessingException e) {
+            final String errorMsg = "An error occurred when generating the JSON response (" + e
+                    .getMessage() + ")";
+            Logger.error(this, e.getMessage(), e);
+            res = ExceptionMapperUtil.createResponse(null, errorMsg);
+        } catch (DotSecurityException e) {
+            final String errorMsg = "The user does not have the required permissions (" + e
+                    .getMessage() + ")";
+            Logger.error(this, e.getMessage(), e);
+            res = ExceptionMapperUtil.createResponse(null, errorMsg);
+        } catch (DotDataException e) {
+            final String errorMsg = "An error occurred when accessing the page information " +
+                    "permissions (" + e.getMessage() + ")";
+            Logger.error(this, e.getMessage(), e);
+            res = ExceptionMapperUtil.createResponse(null, errorMsg);
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)